### PR TITLE
Fix ntp service on SLES11SP4

### PR DESCRIPTION
**Pull request by hjluo**
In SLES11SP4 migration, We'd like to check the ntp service, but the service name
is ntp not ntpd and the SLES11SP4 use the SystemV init.d. To make the service check  
framework to work on SLES11SP4, We'd add some common function to make the service 
actions unified. This means our service check needs to take care of both SystemV
and Systemd services.

- Related ticket: https://progress.opensuse.org/issues/70591
- Needles: N/A
- Verification run: 
https://openqa.nue.suse.com/tests/4854189 
https://openqa.nue.suse.com/tests/4854204 

https://openqa.nue.suse.com/tests/4930320  SLES11sp4
https://openqa.nue.suse.com/tests/4930338  SLES11sp4 
https://openqa.nue.suse.com/tests/4930341  SLES12sp5

http://openqa.nue.suse.com/t4934435 ntp 12sp4 passed
http://openqa.nue.suse.com/t4934470 ntp 12sp5 passed

Regression for console/ntp.pm
https://openqa.nue.suse.com/tests/4940243#step/ntp/38 PASSED

latest 11sp4 run:
https://openqa.nue.suse.com/tests/4940496
https://openqa.nue.suse.com/tests/4940497

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -1825,4 +1827,28 @@ sub install_patterns {
     }
 }
 
+sub common_service_action {
+    my ($service, $type, $action) = @_;
+
+    if ($type eq 'SystemV') {
+        if ($action eq 'on') {
+            assert_script_run 'chkconfig ' . $service . ' on';
+            assert_script_run 'chkconfig ' . $service;
```
Comment (lemon-suse) [inline lib/utils.pm:1836]: It is better to verify the result after the action while not just show the status.

File: lib/utils.pm [hunk 2]
```diff
@@ -1825,4 +1827,28 @@ sub install_patterns {
...
+        } else {
+            assert_script_run '/etc/init.d/' . $service . ' ' . $action;
+            assert_script_run 'service ' . $service . ' status';
```
Comment (lemon-suse) [inline lib/utils.pm:1839]: It is better to verify the result after the action while not just show the status.

File: lib/utils.pm [hunk 3]
```diff
@@ -1825,4 +1827,28 @@ sub install_patterns {
...
+        }
+    }
+    elsif ($type eq 'Systemd') {
+        systemctl $action . ' ' . $service;
+        systemctl 'status ' . $service;
```
Comment (lemon-suse) [inline lib/utils.pm:1844]: It is better to verify the result after the action while not just show the status.

File: lib/service_check.pm [hunk 4]
```diff
@@ -46,10 +46,12 @@ our @EXPORT = qw(
 );
 
 our $hdd_base_version;
-our $support_ver_def  = '11+';
+our $support_ver_def  = '12+';
 our $support_ver_12   = '=12';
 our $support_ver_ge15 = '15+';
 our $support_ver_ge12 = '12+';
+our $support_ver_ge11 = '11+';
+our $support_ver_le12 = '<=12';
```
Comment (lemon-suse) [inline lib/service_check.pm:54]: Have you met a service need several segments for the support_ver? If so, we need enhance here to support such scenario.

File: lib/utils.pm [hunk 5]
```diff
@@ -1825,4 +1827,28 @@ sub install_patterns {
...
+    } else {
+        die "Unsupported service type, please check it again.";
+    }
+}
+
+sub get_service_type {
```
Comment (coolgw) [inline lib/utils.pm:1850]: you have service_type within service_check.pm, but here you create another function?

File: lib/utils.pm (hunk 3 above)
Comment (coolgw) [inline lib/utils.pm:1844]: systemctl 'status ' . $service;  ==> systemctl $action . $service;   ?

File: lib/service_check.pm (hunk 4 above)
Comment (hjluo) [reply lib/service_check.pm:54]: no need till now, for old version of sles, we just have 11sp4 to 12 segment.

File: lib/utils.pm (hunk 1 above)
Comment (coolgw) [inline lib/utils.pm:1836]: this command without any action?

File: lib/utils.pm (hunk 2 above)
Comment (hjluo) [reply lib/utils.pm:1873]: will cook up a common_service_status_check function for this.

File: lib/utils.pm (hunk 3 above)
Comment (hjluo) [reply lib/utils.pm:1878]: I just want to show the status after the 'systemctl action service'

File: lib/utils.pm (hunk 1 above)
Comment (hjluo) [reply lib/utils.pm:1870]: 'on' is special. which need special handle.

File: lib/utils.pm (hunk 5 above)
Comment (hjluo) [reply lib/utils.pm:1884]: changed to use service_check.pm -> module's full function check.

File: lib/utils.pm [hunk 6]
```diff
@@ -1859,4 +1862,24 @@ sub install_patterns {
     }
 }
 
+sub common_service_action {
+    my ($service, $type, $action) = @_;
+
+    if ($type eq 'SystemV') {
+        if ($action eq 'on') {
+            assert_script_run 'chkconfig ' . $service . ' on';
```
Comment (coolgw) [inline lib/utils.pm:1870]: assert_script_run 'chkconfig ' . $service . ' on';   == >      assert_script_run 'chkconfig ' . $service . $action;  ?

File: lib/services/ntpd.pm [hunk 7]
```diff
@@ -38,35 +48,40 @@ sub config_service {
 }
 
 sub enable_service {
-    systemctl('enable ntpd');
+    my ($service, $type) = get_service_name_type();
+    my $cmd = ($type eq 'SystemV') ? 'on' : 'enable';
```
Comment (coolgw) [inline lib/services/ntpd.pm:52]: if you have func common_service_action, then this logic should put into func?

Comment (hjluo) [reply lib/services/ntpd.pm:52]: changed.

File: lib/utils.pm (hunk 6 above)
Comment (hjluo) [reply lib/utils.pm:1870]: for systemV enable and 'start/stop/restart' use different command. I've put the enable
in ntp to the common_service_action function.

File: lib/services/ntpd.pm [hunk 8]
```diff
@@ -18,6 +18,10 @@ use utils;
 use strict;
 use warnings;
 
+# default ntp service type and name
+my $service_type = 'Systemd';
```
Comment (coolgw) [inline lib/services/ntpd.pm:22]: this is needed or not? the parameter should defined in function full_ntpd_check ?

Comment (hjluo) [reply lib/services/ntpd.pm:22]: we have tests/console/ntp.pm called these function. like follow so we'd set default 
service_name and service_type.

sub run {
    select_console 'root-console';
    services::ntpd::install_service();
    services::ntpd::enable_service();

Comment (coolgw) [reply lib/services/ntpd.pm:22]: understand, so we need put logic like：
if (version=11sp4) then {
     service_type=xx
} else {
     service_type=yy
}

**Review by coolgw — APPROVED** [review]

**Comment by lemon-suse** [comment]
LGTM.

**Pull request merged**